### ISSUE TITLE
A way to prevent the default behavior of the keys Enter and Backspace

### ISSUE THREAD

**Issue by DmitrySharabin** [mavoweb/mavo#548]
Working on the [TodoMVC app](https://codepen.io/dsharabin/pen/MWgQNYZ), I faced an issue: for a collection, the `word-break: break-all` rule is set (see line 213 of CSS), so when I press Enter, a new line is added inside a collection element, and after that, the new item is being added to the collection. In the end, I face the situation when the previous element of the collection looks empty, but it's not. After I removed the rule, everything began to work as it should: by pressing Enter, I get a new collection element but not the new line inside the previous item.

Because of the default behavior, I can't implement the feature I need (and already implemented in another place—see lines 7 through 11 of the TodoMVC app code): by pressing Enter in a text field, I trigger the submit event of a form. This feature would let me delete empty collection elements (as the app [specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md#editing) prescribes) by pressing Enter.

So I thought it would be nice if we could prevent the default behavior of the Enter and Backspace keys on a collection (or on an app entirely) the way similar to that we already can disable expressions via the `mv-expressions="none"` and `mv-expressions-ignore` attributes.

Maybe the `mv-enter-ignore` and `mv-backspace-ignore` attributes? Or `mv-keys-ignore="enter, backspace"`? Via the latter, we could specify more precisely which keys to ignore with only one attribute.

I think that feature could also help with David's issue #543.

What do you think?

**Comment by LeaVerou** [comment]
> when I press Enter, a new line is added inside a collection element, and after that, the new item is being added to the collection

That sounds like a bug, though I cannot reproduce it. When I press enter inside a todo item, I just get a new one.

It's unfortunate that the app specification makes bad UX decisions mandatory. Pressing Enter to delete is not externally consistent with any todo app I've seen and is very weird.

Are there real use cases for disabling this behavior? #543 and #544 are bugs, one of which is fixed, and one of which will be, and TodoMVC is a rather niche case of implementing an app based on a spec instead of real needs.
As much as I really want to participate in TodoMVC, we shouldn't add a feature just for that :/

**Comment by DmitrySharabin** [comment]
> That sounds like a bug, though I cannot reproduce it. When I press enter inside a todo item, I just get a new one.

It is because I've already solved that problem by commenting out the corresponding `word-break: break-all` rule in CSS (line 213).

> As much as I really want to participate in TodoMVC, we shouldn't add a feature just for that :/

Absolutely. I think we should close the issue then. :)

Thanks!

**Comment by LeaVerou** [comment]
> Absolutely. I think we should close the issue then. :)

Not necessarily. The line break issue sounds like a bug (enter should EITHER insert a line break OR a new item, but never both), could you make a reduced testcase?

**Comment by DmitrySharabin** [comment]
You know, I can't reproduce it anymore. That's weird. I think that could be an issue with my browser (might be the cache or whatever). Sorry for deceived you.
